Allow solr to start using new SystemV for version 5.2.x

kiwimanman opened the pull request:
Fixed #37 and added integration tests that can be run with `kitchen verify` to make sure solr is running properly after successful convergence. Should be enough to close #46. I feel quite confident that the cookbook will work with the tests in place.

vkhatri commented on .kitchen.yml:
```diff
@@ -1,8 +1,6 @@
 driver:
   name: vagrant
-  network:
-    - ["forwarded_port", {guest: 8983, host: 8888}]
```
i find it useful during `vagrant converge`, any specific reason for not keeping it?

kiwimanman replied:
This gets applied to every vagrant instance kitchen starts and if multiple are run in parallel or if you leave one up while checking another it caused port conflicts in the host.

vkhatri commented on .kitchen.yml:
```diff
@@ -12,10 +10,19 @@ platforms:
   - name: chef/centos-6.5
```
not sure whether it was intentional, but RHEL platform family support is required for vagrant test.

kiwimanman replied:
I never saw RHEL in this list. I'll see if I can add it.

vkhatri commented on recipes/jetty.rb:
```diff
@@ -93,17 +93,29 @@
   end
 end
 
+solr_config_path =  case node['platform_family']
```
i think it would be good to move this to default attributes, like:

```ruby
default['solrcloud']['sysconfig_file'] = value_for_platform_family(
  'debian' => '/etc/default/solr',
  'rhel' => '/etc/sysconfig/solr'
)
```

vkhatri commented on recipes/jetty.rb:
```diff
@@ -93,17 +93,29 @@
+  only_if { node['solrcloud']['version'] > '5.2' }
 end
 
 template '/etc/init.d/solr' do
```
how about calculating source in default attributes? like:

```ruby
if node['solrcloud']['version'] >= '5.2'
  default['solrcloud']['init_source'] = 'v5.2.x/solr.conf.erb'
else
  default['solrcloud']['init_source'] = "#{node['platform_family']}.solr.init.erb"
end
```

i hope template 'v5.2.x/solr.conf.erb' works for both `debian` and `rhel` platform families. will test it for `rhel` anyway.

kiwimanman replied:
I looked through the init.d's for the previous templates and believe it should work.

vkhatri commented on recipes/tarball.rb:
```diff
@@ -24,17 +24,16 @@
 # Require for zk gem
 %w(patch gcc).each do |pkg|
```
these packages install is required during compile for gem `zk`

vkhatri commented on recipes/tarball.rb:
```diff
@@ -24,17 +24,16 @@
-  action :nothing
+  action :install
   only_if { node['solrcloud']['install_zk_gem'] }
-end.run_action(:install)
```
chef_gem `zk` needs to be installed during compile (for `chef11`), or this behavior is replaced with some resource attribute in `chef12`?

kiwimanman replied:
I believe this is carry over from when I was trying to figure out the apt-get issues. It works fine in chef12 but I think you are right about chef11. I'll undo this change.

kiwimanman commented:
I believe this addresses all the feedback. Thanks for being so responsive. The last repo I tried to contribute to was not this pleasant of an experience.

kiwimanman commented:
```
kstone@keistom0:~/chef/chef-solrcloud$ kitchen list
Instance              Driver   Provisioner  Verifier  Transport  Last Action
510-chef-ubuntu-1404  Vagrant  ChefSolo     Busser    Ssh        Verified
510-chef-centos-65    Vagrant  ChefSolo     Busser    Ssh        Verified
520-chef-ubuntu-1404  Vagrant  ChefSolo     Busser    Ssh        Verified
520-chef-centos-65    Vagrant  ChefSolo     Busser    Ssh        Verified
```

vkhatri commented:
@kiwimanman you are welcome! merging this PR.

vkhatri commented:
v5.2.x `solr.conf` has issues, existing `specs` does not capture it at the moment. i will add more specs once i have fixed the issue.

parameter `zkRun=true` is missing and `zkHost` is causing trouble as local zk ensemble never got started. i will check remaining of the parameters.

but PR looks good :+1:  i will adjust few minor changes.

i have merged it locally and should push it out by today. thanks for the PR!
